results-processor/wptreport.py

```diff
@@ -414,6 +414,16 @@ def finalize(self):
         self.sha_product_path
         self.test_run_metadata
 
+    def serialize_gzip(self, filepath):
+        """Serializes and gzips the in-memory report to a file.
+
+        finalize() should be called first.
```

Comment:
What happens otherwise? Can anything be asserted to make sure it has been called?